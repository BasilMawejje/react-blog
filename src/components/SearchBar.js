import React from 'react'
import { connect } from 'react-redux'

import { fetchPosts } from '../actions'

class SearchBar extends React.Component {
    state = { 
        term: ''
     }

    onFormSubmit = (event) => {
        event.preventDefault();
        this.props.onSubmit(this.state.term)
    }

    render(){
        return <div className='ui segment'>
            <form onSubmit={this.onFormSubmit} className='ui form'>
                <div className='field'>
                    <label>Search Posts</label>
                    <input 
                        type='text'
                        value={this.state.term}
                        onChange={e => this.setState({ term: e.target.value })}
                    />
                </div>
            </form>
        </div>
    }
}

const mapStateToProps = state => {
    return { posts: Object.values(state.posts) }
}

export default connect(mapStateToProps, { fetchPosts })(SearchBar);