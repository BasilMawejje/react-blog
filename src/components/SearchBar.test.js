import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import SearchBar from './SearchBar'

configure({ adapter: new Adapter() });

describe('SearchBar', () => {
    const onFormSubmit = jest.fn();
    const onChange = jest.fn();

    let wrapper; 

    beforeEach(() => {
        wrapper = shallow(<SearchBar onChange={onFormSubmit} />);
    });

    it('calls the onFormSubmit function', () => {
        expect(onFormSubmit).not.toHaveBeenCalled();
        expect(onChange).not.toHaveBeenCalled();
        wrapper.find('input').simulate('change', { target: { value: 'mocky' } });
        expect(wrapper.state().term).toBe('mocky');
    });
})