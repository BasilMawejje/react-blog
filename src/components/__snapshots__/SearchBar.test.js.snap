// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders properly 1`] = `
<div
  className="ui segment"
>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        Search Posts
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
